### PR TITLE
sprint-2/issue-#5FomValidation

### LINKED ISSUE #5: Form Validation
The form needs to have validation for each field. 
First Name Input: check the field is not empty
Email input: it needs to check a valid format for an email (hey@hey.com)
Password input: the password needs to be hidden until you press the eye icon on the right side
The user must have to accept the Terms and Privacy Policy checkbox
While these conditions are not valid, the buttons will not change color to allow the sign-up.
Finally, it needs an activity indicator to show if the process was completed successfully.

### PR DESCRIPTION
**Pull request by mcmmiguel**
# Description

What did you implement/Why did you implement this:

- Validations added for each input and for the Terms and Privacy checkbox
- Modal for sign-up implemented

## Screenshots
image image image image

### CODE REVIEW

File: src/components/LoadingModal.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    loadingContainer: {
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+        flex: 1,
+    },
+    successContainer: {
```
Comment (Lizzflores) [inline src/components/LoadingModal.tsx:59]: Hi! its a better practice for you to have a separate file for the styles.

File: src/components/SignUp.tsx [hunk 2]
```diff
@@ -15,54 +15,195 @@ import {
...
+          value={value}
+          onValueChange={handleCheckboxChange}
+          style={value ? styles.checkboxSelected : styles.checkboxUnselected}
+        />
+        <Text>{label}</Text>
+      </View>
+
+      {error ? (<Text style={styles.errorCheckBox}>{error}</Text>) : null}
+
+    </>
```
Comment (Lizzflores) [inline src/components/SignUp.tsx]: Is better to have a file for each component that you're using on your screen. Same thing for the styles, you can have the styles on another file so your code is not to long. Also, you can create another folder within the source folder named screens and place your screens there such as the signup one, login, etc.

**Review by Lizzflores — CHANGES_REQUESTED** [review]

**Review by Lizzflores — COMMENTED** [review]
Additionally, remember to use a tool like 'prettier' to format your code. Also, ensure that all commented lines are in the same language, either Spanish or English and delete the ones you dont need anymore, such as the one with the maxWidth.

File: src/components/LoadingModal.tsx (hunk 1 above)
Comment (DerekAyala) [reply src/components/LoadingModal.tsx:59]: I agree

File: src/interfaces/interfaces.tsx
Comment (DerekAyala) [inline src/interfaces/interfaces.tsx:1]: Great idea separating the interfaces props on a separate file.

**Review by DerekAyala — APPROVED** [review]
LGTM!

File: src/components/LoadingModal.tsx (hunk 1 above)
Comment (Tole15) [reply src/components/LoadingModal.tsx:59]: Yes, no, maybe
I don't know

Comment (Axw3ll) [reply src/components/LoadingModal.tsx:59]: can you repeat the question?

Comment (BrandonVG) [reply src/components/LoadingModal.tsx:59]: You're not the boss of me now

**Review by AlbertoG22 — APPROVED** [review]

File: src/components/SignUp.tsx (hunk 2 above)
Comment (mcmmiguel) [reply src/components/SignUp.tsx]: Requested changes are done on the latest commit

**Review by Lizzflores — APPROVED** [review]

**Review by wiltrouble — APPROVED** [review]

**Pull request merged**